vector/src/main/java/im/vector/app/features/MainActivity.kt

```diff
@@ -170,6 +171,19 @@ class MainActivity : VectorBaseActivity<ActivityMainBinding>(), UnlockedActivity
     }
 
     private fun handleAppStarted() {
+        // On the first run with rust crypto this would be false
+        if (!vectorPreferences.isOnRustCrypto()) {
```

Comment:
I think you need to test `BuildConfig.FLAVOR == "rustCrypto"` also:
```suggestion
        if (BuildConfig.FLAVOR == "rustCrypto" && !vectorPreferences.isOnRustCrypto()) {
```

Reply:
I think it's ok now, isOnRustCrypto will pass to true only if you run once on the new rust flavor. 
If you have a logged in session before that it will store that you had legacy data, and in the check for the popup we do that check:

> vectorPreferences.isOnRustCrypto() && vectorPreferences.hadExistingLegacyData()

Reply:
OK. My point was that the code in the if block will run every time on the KotlinCrypto version. Not a big deal of course.